util/utility.js

```diff
@@ -299,84 +303,70 @@ function generateJob(type, payload) {
  * Injecting API key for Hypixel API
  * Errors from Hypixel API
  * */
-function getData(redis, url, cb) {
-  let u;
-  let delay = Number(config.DEFAULT_DELAY);
-  let timeout = 20000;
-  if (typeof url === 'object' && url && url.url) {
-    u = url.url;
-    delay = url.delay || delay;
-    timeout = url.timeout || timeout;
-  } else {
-    u = url;
+const getData = fromPromise(async (redis, url) => {
+  if (typeof url === 'string') {
+    url = {
+      url,
+    };
   }
-  const parse = urllib.parse(u, true);
-  const hypixelApi = parse.host === 'api.hypixel.net';
-  const mojangApi = parse.host === 'api.mojang.com';
-  const target = urllib.format(parse);
+
+  url = {
+    delay: Number(config.DEFAULT_DELAY),
+    timeout: 20000,
+    retries: 10,
+    ...url,
+  };
+
+  const urlData = urllib.parse(url.url, true);
+  const isHypixelApi = urlData.host === 'api.hypixel.net';
+  const isMojangApi = urlData.host === 'api.mojang.com';
+
+  const target = urllib.format(urlData);
+
   logger.info(`getData: ${target}`);
-  return setTimeout(() => {
-    request({
-      url: target,
-      json: hypixelApi,
-      timeout,
-    }, (err, res, body) => {
-      if (err
-        || !res
-        || res.statusCode !== 200
-        || !body
-        || (hypixelApi && !body.success)
-      ) {
-        // invalid response
-        if (url.noRetry) {
-          return cb('invalid response');
-        }
-        if (mojangApi) {
-          return cb('Failed to get player uuid', null);
-        }
-        if (body) {
-          logger.error(`[INVALID] error: ${body.cause}, retrying ${target}`);
-          if (body.cause === 'Internal error') {
-            return cb('Internal error');
-          }
-        } else {
-          logger.error(`[INVALID] status: ${res ? res.statusCode : ''}, retrying ${target}`);
-        }
-        return getData(redis, url, cb);
-      }
-      if (hypixelApi && !body.success) {
-        // valid response, but invalid data, retry
-        if (url.noRetry) {
-          return cb('invalid data');
-        }
-        // insert errors to redis for monitoring
-        let failed = 0;
-        const multi = redis.multi()
-          .incr('hypixel_api_error')
-          .expireat('hypixel_api_error', getStartOfBlockMinutes(1, 1));
-        multi.exec((err, resp) => {
-          if (err) {
-            logger.error(err);
-          }
-          [failed] = resp;
-          logger.warn(`Failed API requests in the past minute: ${failed}`);
-          logger.error(`[INVALID] data: ${target}, retrying ${JSON.stringify(body)}`);
-          let backoff = (body.throttle)
-            ? 3000
-            : 0;
-          backoff += (failed > 50)
-            ? 10 * failed
-            : 0;
-          logger.debug(`getData timout for ${backoff}ms`);
-          return setTimeout(() => {
-            getData(redis, url, cb);
-          }, backoff);
-        });
-      }
-      return cb(null, body);
+
+  await pTimeout(url.delay);
```

Comment:
After some investigating looks like the tests are failing because you are using `pTimeout` incorrectly. I think using a module here is redundant anyways, since `got` has it's own timeout handling.

Reply:
In that case, I'll remove p-timeout entirely.